python/lsst/meas/algorithms/stamps.py

```diff
@@ -0,0 +1,351 @@
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Collection of small images (stamps).
+"""
+
+__all__ = ["Stamp", "Stamps", "StampsBase", "writeFits", "readFitsWithOptions"]
+
+from collections.abc import Sequence
+import abc
+from dataclasses import dataclass
+import numpy
+
+import lsst.afw.image as afwImage
+import lsst.afw.fits as afwFits
+from lsst.geom import Box2I, Point2I, Extent2I, Angle, degrees, SpherePoint
+from lsst.daf.base import PropertySet
+
+
+def writeFits(filename, stamp_ims, metadata, write_mask, write_variance):
+    """Write a single FITS file containing all stamps.
+
+    Parameters
+    ----------
+    filename : `str`
+        A string indicating the output filename
+    stamps_ims : iterable of `lsst.afw.image.MaskedImageF`
+        An iterable of masked images
+    metadata : `PropertySet`
+        A collection of key, value metadata pairs to be
+        written to the primary header
+    write_mask : `bool`
+        Write the mask data to the output file?
+    write_variance : `bool`
+        Write the variance data to the output file?
+    """
+    metadata['HAS_MASK'] = write_mask
+    metadata['HAS_VARIANCE'] = write_variance
+    metadata['N_STAMPS'] = len(stamp_ims)
+    # create primary HDU with global metadata
+    fitsPrimary = afwFits.Fits(filename, "w")
+    fitsPrimary.createEmpty()
+    fitsPrimary.writeMetadata(metadata)
+    fitsPrimary.closeFile()
+
+    # add all pixel data optionally writing mask and variance information
+    for i, stamp in enumerate(stamp_ims):
+        metadata = PropertySet()
+        metadata.update({'EXTVER': i, 'EXTNAME': 'IMAGE'})
+        stamp.getImage().writeFits(filename, metadata=metadata, mode='a')
+        if write_mask:
+            metadata = PropertySet()
+            metadata.update({'EXTVER': i, 'EXTNAME': 'MASK'})
+            stamp.getMask().writeFits(filename, metadata=metadata, mode='a')
+        if write_variance:
+            metadata = PropertySet()
+            metadata.update({'EXTVER': i, 'EXTNAME': 'VARIANCE'})
+            stamp.getVariance().writeFits(filename, metadata=metadata, mode='a')
+    return None
+
+
+def readFitsWithOptions(filename, stamp_factory, options):
+    """Read stamps from FITS file, allowing for only a
+    subregion of the stamps to be read.
+
+    Parameters
+    ----------
+    filename : `str`
+        A string indicating the file to read
+    stamp_factory : classmethod
+        A factory function defined on a dataclass for constructing
+        stamp objects a la `lsst.meas.alrogithm.Stamp`
+    options : `PropertySet`
+        A collection of parameters.  If certain keys are available
+        (``llcX``, ``llcY``, ``width``, ``height``), a bounding box
+        is constructed and passed to the ``FitsReader`` in order
+        to return a sub-image.
+
+    Returns
+    -------
+    stamps, metadata : `list` of dataclass objects like `Stamp`, PropertySet
```

Comment:
I think we write this as 

```
stamps : `list` of ...
    Some stamps whatever
metadata : `PropertySet
   The metadata
```
rather than a single line.